### ISSUE TITLE
Resize feature for scaled DPI assets

### ISSUE THREAD

**Issue by najdanovicivan** [ImageOptim/ImageOptim#232]
Hi I'd like to request a feature. As an iOS developer i use this app on daily basis to optimize assets. Currently I have to create assets for @3x, @2x and @1x resolution.

I'd like to have feature to use ImageOptim to generate those scaled assets. For example if you input a file named filename@3x.png ImageOptim should output 3 Files:

filename@3x.png - Optimized Image
filename@2x.png - Optimized and scaled to 66.666%
filename.png - Optimized and scaled to 33.333%

If file named filename@2x.png is inputed ImageOptim shoud generate 2 Files:
filename@2x.png - Optimized Image
filename.png - Optimized and scaled to 50%

As far as the UI Goes it can be done with single flag for example like this note the parametere "Resize"

<img width="681" alt="screen shot 2017-08-25 at 16 38 44" src="https://user-images.githubusercontent.com/6421209/29718705-f2b34a2a-89b3-11e7-9875-d015dd688764.png">

**Comment by pornel** [comment]
Currently the architecture assumes one input file = one output file. Changing that would require a bit of refactoring. Would you be able to contribute that?

For example, you'd need to change the table view to be an outline view, in order to group all outputs of the single input file.

**Comment by pathawks** [comment]
What would happen if a smaller file already existed? Would it be replaced? Would it be optimized?

**Comment by monouser7dig** [comment]
If you'd start in a vector based app and export 1x 2x 3x from there ...

....based on the vector base-material I'd argue this yields superior results (compared to resamoling a lossy raster image)  and should thus be the recommended workflow 

After that they can be recompressed in imageoptim

In affinity e.g. you can export 1x 2x 3x with just one click

**Comment by najdanovicivan** [comment]
@pathawks 
If smaller file already exist there is no need for app to do anything about it. ImageOptim should skip generating it. You can always drag it to the queue along with the high res file

**Comment by najdanovicivan** [comment]
@monouser7dig 

I agree about vector images. But what about the raster images. I usually get Images in PSD format and Photoshop isn't very grateful for generating scaled resolutions. You have to export each image 3 times. This is where this feature in ImageOptim will come handy so I'l only have to export @3x assets.

**Comment by najdanovicivan** [comment]
@pornel I'm able to contribute. I'll just need some time to get into the project class structure to see what happens where and when. 

About that table view. Would't it be easier to add rows with file name suffixed with resized and just put the Output file size in the "Size" column and let "Savings" column be always 0% for newly generated files. For example like this:

<img width="604" alt="screen shot 2017-08-28 at 11 13 42" src="https://user-images.githubusercontent.com/6421209/29767318-47ef8d66-8be2-11e7-85e7-f2dd6e0c4e45.png">

**Comment by monouser7dig** [comment]
@najdanovicivan it just works the same for raster images 
One click, three sizes export 

I‘m pretty sure there is an option to get this behaviour in PS as well although it is not a UI tool

**Comment by pornel** [comment]
@najdanovicivan That seems hacky. The files will become separated when you sort the table.

For savings column you can use `nil`, so it'll be just empty.

**Comment by amanpatel** [comment]
+1  Resizing would help photographers too. Some of us host web size files on a gallery and the full size on a media or online hosting service. Having ImageOptim resize + optimize in 1 step would simplify workflow.
